geojson_pydantic/geometries.py

```diff
@@ -27,6 +28,18 @@ class _GeometryBase(BaseModel, abc.ABC):
     def __geo_interface__(self):
         return self.dict()
 
+    @classmethod
+    def validate(cls, value):
+        try:
+            value = json.loads(value)
+        except TypeError:
+            try:
+                return cls(**value.dict())
```

Comment:
Because we can create a `GeometryCollection` with `Geometry` as arguments, which has a `List[Geometry]` as attribute (`Geometry = Union[Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon].`)

The validation of the `geometries: List[Geometry]` attribute is run with an object of type `Geometry`. Such object cannot be casted into a sublass of _GeometryBase itself, but since it's a subclass to Pydantic, we can just call `.dict` to get the value as a dict.

Reply:
And I prefer the EAFP principle instead of LYBL that would had required a check of instance of `BaseModel`.

Reply:
Cool that makes sense to me, thanks for the explanation!